packages/wrangler/CONTRIBUTING.md

```diff
@@ -0,0 +1,156 @@
+# Wrangler Contributing Guidelines for Internal Teams
+
+## What you will learn
+
+- How to add a new command to Wrangler
+- How to specify arguments for a command
+- How to use experimental flags
+- How to read from the config
+- How to implement a command
+- How to test a command
+
+## Defining your new command to Wrangler
+
+1. define the command with the util defineCommand
+
+```ts
+import { defineCommand, defineNamespace } from "./util";
+
+// Namespaces are the prefix before the subcommand
+// eg "wrangler kv" in "wrangler kv put"
+// eg "wrangler kv key" in "wrangler kv key put"
+defineNamespace({
+	command: "wrangler kv",
+	metadata: {
+		description: "Commands for interacting with Workers KV",
+		status: "stable",
+	},
+});
+// Every level of namespaces must be defined
+// eg "wrangler kv key" in "wrangler kv key put"
+defineNamespace({
+	command: "wrangler kv key",
+	metadata: {
+		description: "Commands for interacting with Workers KV data",
+		status: "stable",
+	},
+});
+
+// Define the command args, implementation and metadata
+const command = defineCommand({
+	command: "wrangler kv key put", // the full command including the namespace
+	metadata: {
+		description: "Put a key-value pair into a Workers KV namespace",
+		status: "stable",
+	},
+	args: {
+		key: {
+			type: "string",
+			description: "The key to put into the KV namespace",
+			required: true,
+		},
+		value: {
+			type: "string",
+			description: "The value to put into the KV namespace",
+			required: true,
+		},
+		"namespace-id": {
+			type: "string",
+			description: "The namespace to put the key-value pair into",
+			required: true,
+		},
+	},
+	// the positionalArgs defines which of the args are positional and in what order
+	positionalArgs: ["key", "value"],
+	handler(args) {
+		// implementation here
+	},
+});
+```
+
+2. global vs shared vs command specific (named + positional) args
+
+- Command-specific args are defined in the `args` field of the command definition. Command handlers receive these as a typed object automatically. To make any of these positional, add the key to the `positionalArgs` array.
+- You can share args between commands by declaring a separate object and spreading it into the `args` field. Feel free to import from another file.
+- Global args are shared across all commands and defined in `src/commands/global-args.ts` (same schema as command-specific args). They are passed to every command handler.
+
+3. (get a type for the args)
+
+You may want to pass your args to other functions. These functions will need to be typed. To get a type of your args, you can use `typeof command.args`.
+
+4. implement the command handler
+
+A command handler is just a function that receives the args as the first param. This is where you will want to do API calls, I/O, logging, etc.
+
+- api calls
+
+Define API response type. Use `fetchResult` to make API calls. `fetchResult` will throw an error if the response is not 2xx.
```

Comment:
This should link to where to import `fetchResult()` from, and should mention that it makes authed requests to the Cloudflare API.